packages/shared_preferences/lib/shared_preferences.dart

```diff
@@ -74,9 +80,11 @@ class SharedPreferences {
   /// exception if it's not a string set.
   List<String> getStringList(String key) {
     List<Object> list = _preferenceCache[key];
-    if (list != null && list is! List<String>) {
-      list = list.cast<String>().toList();
-      _preferenceCache[key] = list;
+    if (list != null) {
+      list = List<String>.from(list);
+      if (list is! List<String>) {
```

Comment:
Can this `if` condition ever be true? On line 84, we assign `list = List<String>.from(list)`, so is it even possible for `list` to be anything else than `List<String>` on line 85?

Reply:
Looks like an overlook on my part. Sorry! Will issue a fix.

Reply:
@apaatsio Does it look better now?